Framework/API/test/ResultTest.h

```diff
@@ -0,0 +1,49 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#pragma once
+
+#include <cxxtest/TestSuite.h>
+
+#include <MantidAPI/Result.h>
+
+using namespace Mantid::API;
+
+class ResultTest : public CxxTest::TestSuite {
+public:
+  void testStringErrors() {
```

Comment:
I would also be tempted to split out the tests into those that test success paths and those that test the error paths and consider [naming](https://kevlinhenney.medium.com/a-test-of-knowledge-78f4688dc9cb) them a little more verbosely to indicate what it means if the tests fail.

I'm happy with the current changes if this needs to be done for a code freeze or release window and this change can be done afterward.